File: fapi/src/main/java/com/artsykov/fapi/models/SubscriptionModel.java
package com.artsykov.fapi.models;

import lombok.Data;
import org.springframework.stereotype.Component;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import java.sql.Date;
import java.sql.Timestamp;

@Data
public class SubscriptionModel {
    @Min(value=0, message="Incorrect subscription id")
    private int id;
    private ProductModel product;

    @Min(value=1, message="Incorrect customer id")
    private int idCustomer;
    private Date startSubscriptionDate;

    @Min(value=0, message="Incorrect active status")
    @Max(value=1, message="Incorrect active status")
    private int isActive;

    public SubscriptionModel() {}

    public SubscriptionModel(int id, ProductModel product, int idCustomer, Date startSubscriptionDate, int isActive) {
        this.id = id;
        this.product = product;
        this.idCustomer = idCustomer;
        this.startSubscriptionDate = startSubscriptionDate;
        this.isActive = isActive;
    }
}

File: frontend/src/app/modules/models/subscriptionModel.ts
import {ProductModel} from "./productModel";

export class subscriptionModel {
  id: number;
  product: ProductModel;
  idCustomer: number;
  startSubscriptionDate: Date;
  isActive: number;

  static cloneBase(subscription: subscriptionModel): subscriptionModel {
    const clonedSubscription: subscriptionModel = new subscriptionModel();
    clonedSubscription.id = subscription.id;
    clonedSubscription.product = subscription.product;
    clonedSubscription.idCustomer = subscription.idCustomer;
    clonedSubscription.startSubscriptionDate = subscription.startSubscriptionDate;
    clonedSubscription.isActive = subscription.isActive;
    return clonedSubscription;
  }
}
